log4j-core/src/main/java/org/apache/logging/log4j/core/async/DisruptorUtil.java

```diff
@@ -153,4 +155,43 @@ public Long call() {
             throw new IllegalStateException(msg, ex);
         }
     }
+
+    /**
+     * Creates a new {@link Semaphore} or null based on
+     * <pre>AsyncLogger.AsyncQueueFullEnqueueSemaphorePermits</pre>.
+     */
+    static Semaphore createAsyncLoggerQueueFullEnqueuSemaphore() {
+        return createAsyncQueueFullEnqueuSemaphore("AsyncLogger.AsyncQueueFullEnqueueSemaphorePermits");
+    }
+
+    /**
+     * Creates a new {@link Semaphore} or null based on
+     * <pre>AsyncLoggerConfig.AsyncQueueFullEnqueueSemaphorePermits</pre>.
+     */
+    static Semaphore createAsyncConfigQueueFullEnqueuSemaphore() {
+        return createAsyncQueueFullEnqueuSemaphore("AsyncLoggerConfig.AsyncQueueFullEnqueueSemaphorePermits");
+    }
+
+    /**
+     * LOG4J2-2606: Disruptor spins enqueuing events across multiple threads when the queue is full.
+     * CPU utilization is significantly reduced by restricting access to the enqueue operation.
+     * Non-positive values disable the semaphore.
+     */
+    private static Semaphore createAsyncQueueFullEnqueuSemaphore(String property) {
```

Comment:
Nit: ...EnqueuSemaphore -> ...EnqueueSemaphore

Also, I'm not sure that it is accurate to say "Disruptor spins", that makes it sound like this is by design. What about something like this:
> LOG4J2-2606: users encountered excessive CPU utilization with Disruptor v3.4.2 when the application
> was logging more than the underlying appender could keep up with and the ringbuffer became full,
> especially when the number of application threads vastly outnumbered the number of cores.
> CPU utilization is significantly reduced by...

Reply:
> Nit: ...EnqueuSemaphore -> ...EnqueueSemaphore

Oops, that's embarrassing! Fixed and updated the comment based on your suggestion.